.github/workflows/lint.yml

```diff
@@ -0,0 +1,44 @@
+on:
```

Comment:
@mikiquantum this file is basically a copy of `tests.yml` with only linting related stuff. Is there a way of drying the common pieces between this, `tests.yml`, `build.yml`, etc?

Reply:
I think is fine, we could start using reusable workflows but that would most likely mean that we need to refactor a bit the current ones.